unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentView.cs

```diff
@@ -0,0 +1,196 @@
+    {
+        base.Initialize();
+        Configure(model);
+    }
+
+    public void Configure(GridContainerComponentModel model)
+    {
+        this.model = model;
+        RefreshControl();
+    }
+
+    public override void RefreshControl()
+    {
+        if (model == null)
+            return;
+
+        SetNumColumns(model.numColumns);
+        SetItemSize(model.itemSize);
+        SetSpaceBetweenItems(model.spaceBetweenItems);
+        SetItems(model.items);
+    }
+
+    public override void Dispose()
+    {
+        base.Dispose();
+
+        RemoveAllInstantiatedItems();
+    }
+
+    public void SetNumColumns(int newNumColumns)
+    {
+        model.numColumns = newNumColumns;
+
+        if (gridLayoutGroup == null)
+            return;
+
+        gridLayoutGroup.constraintCount = newNumColumns;
+    }
+
+    public void SetItemSize(Vector2 newItemSize)
+    {
+        model.itemSize = newItemSize;
+
+        if (gridLayoutGroup == null)
+            return;
+
+        gridLayoutGroup.cellSize = newItemSize;
+    }
+
+    public void SetSpaceBetweenItems(Vector2 newSpace)
+    {
+        model.spaceBetweenItems = newSpace;
+
+        if (gridLayoutGroup == null)
+            return;
+
+        gridLayoutGroup.spacing = newSpace;
+    }
+
+    public void SetItems(List<BaseComponentView> items)
+    {
+        model.items = items;
+
+        RemoveAllInstantiatedItems();
+
+        for (int i = 0; i < items.Count; i++)
+        {
+            CreateItem(items[i], $"Item{i}");
+        }
+    }
+
+    public BaseComponentView GetItem(int index)
+    {
+        if (index >= instantiatedItems.Count)
+            return null;
+
+        return instantiatedItems[index];
+    }
+
+    public List<BaseComponentView> GetAllItems() { return instantiatedItems; }
+
+    internal void CreateItem(BaseComponentView newItem, string name)
+    {
+        if (Application.isPlaying)
+        {
+            InstantiateItem(newItem, name);
+        }
+        else
+        {
+            if (isActiveAndEnabled)
+                StartCoroutine(IntantiateItemOnEditor(newItem, name));
+        }
+    }
+
+    internal void InstantiateItem(BaseComponentView newItem, string name)
+    {
+        if (newItem == null)
+            return;
+
+        BaseComponentView newGO = Instantiate(newItem, transform);
+        newGO.name = name;
+        instantiatedItems.Add(newGO);
+    }
+
+    internal IEnumerator IntantiateItemOnEditor(BaseComponentView newItem, string name)
+    {
+        yield return null;
+        InstantiateItem(newItem, name);
+    }
+
+    internal void RemoveAllInstantiatedItems()
+    {
+        foreach (Transform child in transform)
+        {
+            if (Application.isPlaying)
+            {
+                Destroy(child.gameObject);
+            }
+            else
+            {
+                if (isActiveAndEnabled)
+                    StartCoroutine(DestroyGameObjectOnEditor(child.gameObject));
```

Comment:
I had to separate the instantiation/destruction of items between play and editor modes because (I don't really understand the reason) when I refreshed the UI control from Unity editor (all UI components have a REFRESH button in its inspector to see the model changes reflected in the scene view), it seems the editor was not be able to reflect the change instantly just after we pressed the REFRESH button, however if we waited for a frame the changes were reflected in the scene correctly.

![image](https://user-images.githubusercontent.com/64659061/134487181-a30cf161-ae06-4516-a93f-6a16d8575fc1.png)